lib/origin_simulator/payload.ex

```diff
@@ -1,38 +1,34 @@
 defmodule OriginSimulator.Payload do
   use GenServer
 
-  alias OriginSimulator.{Body, Recipe, Size}
+  alias OriginSimulator.{Body, Recipe}
 
   ## Client API
 
   def start_link(opts) do
-    GenServer.start_link(__MODULE__, opts, name: :payload)
+    GenServer.start_link(__MODULE__, opts, name: :payloads)
   end
 
-  def fetch(server, %Recipe{origin: value}) when is_binary(value) do
-    GenServer.call(server, {:fetch, value})
+  def update_recipe_book(server, recipe_book) do
+    GenServer.call(server, {:update_recipe_book, recipe_book})
   end
 
-  def fetch(server, %Recipe{body: value}) when is_binary(value) do
-    GenServer.call(server, {:parse, value})
-  end
-
-  def fetch(server, %Recipe{random_content: value}) when is_binary(value) do
-    GenServer.call(server, {:generate, value})
-  end
-
-  def body(_server, status) do
+  def body(_server, status, route) do
     case status do
-      200 -> cache_lookup()
+      200 -> cache_lookup(route)
       404 -> {:ok, "Not found"}
       406 -> {:ok, "Recipe not set, please POST a recipe to /add_recipe"}
       _   -> {:ok, "Error #{status}"}
     end
   end
 
-  defp cache_lookup() do
-    case :ets.lookup(:payload, "body") do
-      [{"body", body}] -> {:ok, body}
+  def restart do
+    GenServer.stop(:payloads)
+  end
+
+  defp cache_lookup(route) do
+    case :ets.lookup(:payloads, route) do
+      [{^route, body}] -> {:ok, body}
```

Comment:
Currently, our single route acts as a catch-all. 
You are matching by value with the pin operator which will only match exact routes